test/parallel/test-worker-no-sab.js

```diff
@@ -0,0 +1,17 @@
+// Flags: --no-harmony-sharedarraybuffer
+
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { isMainThread, Worker } = require('worker_threads');
+
+// Regression test for https://github.com/nodejs/node/issues/39717.
+
+const w = new Worker(__filename);
+
+w.on('exit', common.mustCall((status) => {
+  assert.strictEqual(status, 2);
+}));
+
+if (!isMainThread) process.exit(2);
```

Comment:
```suggestion
const { Worker } = require('worker_threads');

// Regression test for https://github.com/nodejs/node/issues/39717.

// Do not use isMainThread so that this test itself can be run inside a Worker.
if (!process.env.HAS_STARTED_WORKER) {
  process.env.HAS_STARTED_WORKER = 1;
  const w = new Worker(__filename);

  w.on('exit', common.mustCall((status) => {
    assert.strictEqual(status, 2);
  }));
} else {
  process.exit(2);
}
```

Reply:
@codebytere Are you ok if we commit this suggestion, run tests, check with @BridgeAR if they can then approve the PR,, and hopefully land? Or is there something about this suggestion that would make you reluctant to do that?